### DISCUSSION TITLE
How to compile https://github.com/ROCmSoftwarePlatform/bitsandbytes?

### DISCUSSION THREAD

**Discussion by liangyong928** [ROCm/ROCm#3120] [Q&A] [answered]
I saw on the [ROCm blogs](https://rocm.blogs.amd.com/) that to install bitsandbytes for ROCm GPUs:
For ROCm 6.0+
```
# Install bitsandbytes
git clone --recurse https://github.com/ROCm/bitsandbytes.git
cd bitsandbytes
git checkout rocm_enabled
make hip
python setup.py install
```
However, the rocm_enabled branch of bitsandbytes does not seem to run the make hip command:
```
$ make hip
make: *** No rule to make target 'hip'. Stop.
```
I also tried compiling according to the README at https://github.com/ROCm/bitsandbytes.git, but that wasn't successful either.

How can I compile https://github.com/ROCmSoftwarePlatform/bitsandbytes?

**Comment by garrettbyrd** [comment, marked as answer]
There are `Makefile`s in other branches. E.g., `rocm6.2_internal_testing` ([link](https://github.com/ROCm/bitsandbytes/tree/rocm6.2_internal_testing)), although I don't think dropping it straight into the `rocm_enabled` branch works. I know you'd have to change line 30 from `FILES_CPP := $(CSRC)/common.cpp $(CSRC)/cpu_ops.cpp $(CSRC)/pythonInterface.c` to `FILES_CPP := $(CSRC)/common.cpp $(CSRC)/cpu_ops.cpp $(CSRC)/pythonInterface.cpp`, since that file is different between these branches. There are likely other tweaks you'd have to make as well, but I'm also currently working through this.

Why there is no `Makefile` in the `rocm_enabled` branch though, I have no idea.

**Comment by garrettbyrd** [comment]
@liangyong928 I wrote up a more thorough explanation of what's going on [here](https://github.com/ROCm/ROCm/issues/3132). Unsurprisingly, it is mainly a documentation issue. Assuming you're on ROCm 6.x, you'll want to be using the branch `rocm6.2_internal_testing`.
